src/python_testing/TC_CADMIN_1_19.py

```diff
@@ -0,0 +1,175 @@
+        initial_number_of_fabrics = len(fabrics)
+
+        self.step(4)
+        OC_cluster = Clusters.OperationalCredentials
+        max_fabrics = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.SupportedFabrics)
+
+        self.step(5)
+        fids_ca_dir = {}
+        fids_fa_dir = {}
+        fids = {}
+        self.print_step("total fabrics", max_fabrics - initial_number_of_fabrics)
+        for fid in range(1, max_fabrics - initial_number_of_fabrics):
+            # Make sure that current test step is 5, resets here after each loop
+            self.current_step_index = 5
+
+            self.step("5a")
+            params = await self.openCommissioningWindow(dev_ctrl=self.th1, timeout=self.max_window_duration, node_id=self.dut_node_id)
+
+            self.step("5b")
+            fids_ca_dir[fid] = self.certificate_authority_manager.NewCertificateAuthority()
+            fids_fa_dir[fid] = fids_ca_dir[fid].NewFabricAdmin(vendorId=0xFFF1, fabricId=fid)
+            fids[fid] = fids_fa_dir[fid].NewController(nodeId=fid)
+
+            await fids[fid].CommissionOnNetwork(
+                nodeId=self.dut_node_id, setupPinCode=params.commissioningParameters.setupPinCode,
+                filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=params.randomDiscriminator)
+
+            self.print_step("commissioning iteration", fid)
+
+        self.step(6)
+        # TH reads the CommissionedFabrics attributes from the Node Operational Credentials cluster
+        current_fabrics = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.SupportedFabrics)
+        if current_fabrics != max_fabrics:
+            asserts.fail(f"Expected number of fabrics not correct, instead was {current_fabrics}")
+
+        self.step(7)
+        params = await self.openCommissioningWindow(dev_ctrl=self.th1, node_id=self.dut_node_id)
+
+        self.step(8)
+        # TH creates a controller on a new fabric and commissions DUT_CE using that controller
+        fids_ca_dir[current_fabrics + 1] = self.certificate_authority_manager.NewCertificateAuthority()
+        fids_fa_dir[current_fabrics + 1] = fids_ca_dir[current_fabrics +
+                                                       1].NewFabricAdmin(vendorId=0xFFF1, fabricId=current_fabrics + 1)
+        try:
+            fids[current_fabrics + 1] = fids_fa_dir[current_fabrics + 1].NewController(nodeId=current_fabrics + 1)
+            await fids[current_fabrics + 1].CommissionOnNetwork(
+                nodeId=self.dut_node_id, setupPinCode=params.commissioningParameters.setupPinCode,
+                filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=params.randomDiscriminator)
+
+        except ChipStackError as e:
+            self.print_step("err", e.err)
```

Comment:
remove print - think this might be leftover debug stuff.

Reply:
thank you, I have removed this debugging print statement.